scripts/tasks_retrieval/Reactome_tasks_creation_full_set.py

```diff
@@ -0,0 +1,371 @@
+            verbose (bool, optional): If true warnings will be printed. Defaults to False.
+
+        Returns:
+        -------
+            list[str]: a list of gene symbols
+
+        """
+        sub_genes = set()
+        if pathway in self.pathway_to_gene:
+            return self.pathway_to_gene[pathway]
+        elif not pathway in self.hierarchies:
+            if verbose:
+                warnings.warn(
+                    f"Pathway {pathway} has no sub pathways and no genes defined"
+                )
+        else:
+            for sub_pathways in self.hierarchies[pathway]:
+                sub_genes.update(self.get_genes(sub_pathways))
+        self.pathway_to_gene[pathway] = list(sub_genes)
+        return list(sub_genes)
+
+
+@click.command()
+@click.option(
+    "--main-task-directory",
+    "-m",
+    type=click.STRING,
+    help="The task root directory.  Will not be created.",
+    default="./tasks",
+)
+@click.option(
+    "--allow-downloads",
+    type=click.BOOL,
+    help="If false data files will be downloaded directly from HGNC and reactome, set to true only if you trust the urls above",
+    default=False,
+)
+@click.option(
+    "--pathway-identifier",
+    "-p",
+    type=click.STRING,
+    help="Pathway identifier from which we want to create multilabel task.",
+    multiple=True,
+    default=None,
+)
+@click.option(
+    "--hierarchy-file",
+    type=click.STRING,
+    help="The location of the ReactomePathwaysRelation file available at https://reactome.org/download-data",
+    default=None,
+)
+@click.option(
+    "--pathway-description",
+    type=click.STRING,
+    help="The location of the ReactomePathways.txt file available at https://reactome.org/download-data",
+    default=None,
+)
+@click.option(
+    "--pathway-to-gene-file",
+    "-m",
+    type=click.STRING,
+    help="A file with the bottom pathways and the genes involved in the pathways available at https://reactome.org/download-data",
+    default=None,
+)
+@click.option(
+    "--verbose/--quite",
+    "-v/-q",
+    is_flag=True,
+    default=True,
+)
+@click.option(
+    "--add-top-levels",
+    help="If true will create a task per all of the top level human pathways and per each top level pathway individually for Homo sapiens",
+    type=click.BOOL,
+    default=True,
+)
+def main(
+    main_task_directory,
+    allow_downloads,
+    pathway_identifier,
+    hierarchy_file,
+    pathway_description,
+    pathway_to_gene_file,
+    verbose,
+    add_top_levels,
+):
+
+    hierarchies_file_url = verify_source_of_data(
+        hierarchy_file, url=HIERARCHIES_URL, allow_downloads=allow_downloads
+    )
+
+    pathway_to_gene_url = verify_source_of_data(
+        pathway_to_gene_file,
+        url=BOTTOM_PATHWAY_TO_GENE_LIST,
+        allow_downloads=allow_downloads,
+    )
+    pathway_des_url = verify_source_of_data(
+        pathway_description, url=PATHWAY_DES, allow_downloads=allow_downloads
+    )
```

Comment:
i think we should create a dictionary for the urls and the files, and then iterate over them. The current code is repetitive.